[Bug]: Plugin crashes with "can't use ElementIterator on unknown value" when a hcloud_server's dynamic network block has an unknown network_id

### What happened?

A `terraform apply` that creates an `hcloud_network` and an `hcloud_server` attached to it *in the same apply* crashes the provider outright instead of producing a normal plan/diff:

```
╷
│ Error: Plugin did not respond
│
│   with hcloud_server.this,
│   on main.tf line X, in resource "hcloud_server" "this":
│    X: resource "hcloud_server" "this" {
│
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).PlanResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-hcloud_v1.66.0 plugin:

panic: can't use ElementIterator on unknown value

goroutine ...:
github.com/hetznercloud/terraform-provider-hcloud/internal/server.validateUniqueNetworkIDs(...)
	internal/server/resource.go:1495
github.com/hetznercloud/terraform-provider-hcloud/internal/server.resourceServerCustomizeDiff(...)
	internal/server/resource.go:1489
...
```

Root-caused by reading the provider's own source (not guessed). `internal/server/resource.go`'s `resourceServerCustomizeDiff` calls `validateUniqueNetworkIDs(d)`:

```go
func validateUniqueNetworkIDs(d *schema.ResourceDiff) error {
	if rawNetworks := d.GetRawConfig().GetAttr("network"); !rawNetworks.IsNull() {
		for it := rawNetworks.ElementIterator(); it.Next(); {
			...
```

This only guards against `rawNetworks` being **null** — it never checks whether the value is **known**. When a `network { network_id = ... }` block references a resource attribute that hasn't been created yet in the same apply (e.g. `hcloud_network.this.id`, where `hcloud_network.this` doesn't exist in state yet), the entire `network` set becomes a wholly *unknown* `cty.Value` — not null, but unknown — and `cty.Value.ElementIterator()` panics unconditionally on an unknown value (documented `go-cty` behavior, not itself a bug). The missing guard is an `IsWhollyKnown()` (or equivalent) check before iterating.

This isn't specific to any one config shape — it reproduces for **any** `hcloud_server` whose `network.network_id` isn't already a known value at plan time, which is the normal case for a from-scratch `terraform apply` creating the network and the server together.

Confirmed live against `terraform-provider-hcloud` v1.66.0 (latest as of 2026-07-14), reproduced consistently across multiple fresh applies — not a stale-version issue. Terraform >= 1.9.

### What did you expect to happen?

The plan/apply should succeed normally (or, if the provider genuinely can't validate uniqueness against an unknown value, it should skip that validation for this plan and defer it to a later one, not panic). A first-time `terraform apply` that creates a network and attaches a server to it in the same run is a completely ordinary Terraform pattern and shouldn't crash the provider.

**Workaround that does work**, for anyone finding this issue: apply in two passes —

```
terraform apply -target=hcloud_network.this -target=hcloud_network_subnet.this
terraform apply
```

Once the network exists in state, `network_id` is known on the second apply and the crash doesn't occur. Every subsequent apply against existing state is unaffected — this only hits the very first bootstrap.

**Suggested fix**: add an `IsWhollyKnown()` (or `IsKnown()`) guard in `validateUniqueNetworkIDs` before calling `ElementIterator()`, and skip validation for this plan rather than panicking when the value isn't known yet. Looks like a one-line guard clause — happy to open a PR if that's useful.


### Please provide a minimal working example


```hcl
terraform {
  required_providers {
    hcloud = {
      source  = "hetznercloud/hcloud"
      version = "~> 1.66"
    }
  }
}

provider "hcloud" {
  token = var.hcloud_token
}

variable "hcloud_token" {
  type      = string
  sensitive = true
}

resource "hcloud_network" "this" {
  name     = "example"
  ip_range = "10.0.0.0/16"
}

resource "hcloud_network_subnet" "this" {
  network_id   = hcloud_network.this.id
  type         = "cloud"
  network_zone = "eu-central"
  ip_range     = "10.0.1.0/24"
}

resource "hcloud_server" "this" {
  name        = "example"
  server_type = "cx23"
  image       = "alma-9"
  location    = "fsn1"

  network {
    network_id = hcloud_network.this.id
  }

  depends_on = [hcloud_network_subnet.this]
}
```

Run `terraform apply` from a clean state — the crash happens on the very first apply, before either resource exists.

--- a/internal/server/resource.go
+++ b/internal/server/resource.go
@@ -1492,7 +1492,7 @@ func resourceServerCustomizeDiff(_ context.Context, d *schema.ResourceDiff, _ an
 
 func validateUniqueNetworkIDs(d *schema.ResourceDiff) error {
 	// Validate that at least one of network_id or subnet_id is specified.
-	if rawNetworks := d.GetRawConfig().GetAttr("network"); !rawNetworks.IsNull() {
+	if rawNetworks := d.GetRawConfig().GetAttr("network"); rawNetworks.IsWhollyKnown() && !rawNetworks.IsNull() {
 		for it := rawNetworks.ElementIterator(); it.Next(); {
 			_, networkVal := it.Element()
 

